components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -601,7 +601,7 @@ public void addResidentIdP(IdentityProvider identityProvider, String tenantDomai
 
         IdentityProviderProperty sessionIdletimeOutProperty = new IdentityProviderProperty();
         String idleTimeout = IdentityUtil.getProperty(IdentityConstants.ServerConfig.SESSION_IDLE_TIMEOUT);
-        if (StringUtils.isBlank(idleTimeout) || !StringUtils.isNumeric(rememberMeTimeout)) {
+        if (StringUtils.isBlank(idleTimeout) || !StringUtils.isNumeric(idleTimeout)) {
```

Comment:
Haven't checked for < 0. What you can do is use Integer.parseInt and check > 0. Since you've already verified isNumeric you can't get an exception.